jck/diff/diff.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash 
+
+setup()
+{
+	cd $WORKDIR
+	version=$1
+	repo=$2
+	
+	if [ -d "$WORKDIR/$version" ] ; then 
+		echo "Using existing test repo at: $WORKDIR/$version"
+	else 
+		echo "Git cloning test materials from $repo..."
+		git clone $repo $version
+		if [[ $? != 0 ]]; then 
+			exit $rc; 
+		fi
+	fi 
+}
+
+compare()
+{
+	rm -rf $WORKDIR/*.lst
+	rm -rf $WORKDIR/*.log
+
+	dirName=$1
+
+	cd $WORKDIR/$VERSION1/JCK-$dirName-$VERSION_VALUE1/tests
+	echo "Listing test directories under $dirName at: `pwd`" 
+	find . -maxdepth 2 -mindepth 2 -type d > $WORKDIR/$dirName-$VERSION_VALUE1.lst
+	
+	cd $WORKDIR/$VERSION2/JCK-$dirName-$VERSION_VALUE2/tests
+	echo "Listing test directories under $dirName at: `pwd`" 
+	find . -maxdepth 2 -mindepth 2 -type d > $WORKDIR/$dirName-$VERSION_VALUE2.lst
+
+
+	if cmp -s $WORKDIR/$dirName-$VERSION_VALUE1.lst $WORKDIR/$dirName-$VERSION_VALUE2.lst; then 
+		echo "SAME : '$dirName' folder identical in both given versions: $VERSION1 & $VERSION2"
+	else 
+		diff $WORKDIR/$dirName-$VERSION_VALUE1.lst $WORKDIR/$dirName-$VERSION_VALUE2.lst > $WORKDIR/$dirName-diff.log 
+		echo "DIFFERENT : '$dirName' folder content are different in two given versions: $VERSION1 & $VERSION2"
+		echo "Please manually investigate the following differences in the two given repositories:"
+		cat $WORKDIR/$dirName-diff.log
+	fi 
+
+	rm -rf $WORKDIR/*.lst
```

Comment:
We may want to keep it for debugging purposes.

Reply:
Moved that line inside setup(), so the *.lst files will remain in workspace after the process completes.

Reply:
Actually, I've changed the code further after the above comment. The script is no longer generating any *.lst files. All text files it generates are *.txt (easily viewable from jenkins build page when archived).